### ISSUE TITLE
"King captures rook" is poor UI gesture for castling

### ISSUE THREAD

**Issue by mcostalba** [cutechess/cutechess#97]
"King captures rook" is an internal / technical choice to codify the castling but IMO when playing a game, the natural choice for castling gesture is to resemble the actual real board and the actual real gesture you do on the wood board. So for castling user should place his kingin his final destination and not above the rook.

**Comment by ilaripih** [comment]
The real gesture would of course be to move both the king and the rook, but that's not very convenient for a chess UI. "King captures rook" was chosen because it's unambiguous even in random variants where the king may have two different moves with the same target square.

I just checked lichess' (which supports Chess960) implementation: they have both "king captures rook" and "king moves to target square" allowed at the same time (both squares highlighted as legal targets). Would that be better?

Or maybe we could keep the "king captures rook" implementation but make it more clear what's going to happen when that move is made by showing with arrows where the king and rook are going to move. Then it wouldn't be possible to make a normal move by accident when intending the castle.

**Comment by mcostalba** [comment]
Yes, I understand to special case the standard chess may not seem
attractive to a developer :-) just please  evaluate if it would make sense
considering this is the chess played in the vast majority of cases.

**Comment by lucasart** [comment]
I like it the way it is :) When you hover your mouse over the king, the available target squares are shown. I don't think arrows are a good idea, as they would be quite invasive visually.

**Comment by ilaripih** [comment]
Thanks for the input. The biggest difficulty in UI design is that people like different things. I think I'll go the lichess.org way as they have clearly put a lot of thought into their UI.

**Comment by d3vv** [comment]
As I saw many people don't understand how to make castling into Cutechess GUI (just click on a rook).
Sure "Castling consists of moving the king two squares towards a rook" a better way for most.

**Comment by alwey** [comment]
When I first used (and experimented with) CuteChess GUI I could not make castling work immediately but with square highlighting one can figure this out within a minute. I actually implemented some aspects of this feature request some time ago but witheld the results. We are quite possibly going to run into problems with variant play and break things.

**Comment by Rescator7** [comment]
The way to castle here is completly wrong.
All softwares about chess does it right you have to move the king by 2 squares. The rook is then move by the software.
In real life, official rules, you have to make the move by 1 hand, and move the king by two square first, then move the rook. If you touch the rook first, you must play the rook! Doing castle by two hands is also prohibited.
Please fix that. It's ugly.

**Comment by alwey** [comment]
@Rescator7: Touch-move has never applied for the GUI of Cutechess. Only when you release the piece the move is entered.
